### DISCUSSION TITLE
Update Message

### DISCUSSION THREAD

**Discussion by dadams39** [microsoftgraph/msgraph-sdk-go#156] [Q&A]
I did not see this feature in graph explorer, but I was curious if the SDK has the capability. 
If I have a message object in memory and it has been deleted by an illegal user, is it possible to put the message back into the mailbox? 

If it is possible what is the generic command that would make that possible?

**Comment by baywet** [comment]
you can [create a draft message](https://docs.microsoft.com/en-us/graph/api/user-post-messages?view=graph-rest-1.0&tabs=http) and later on [send it](https://docs.microsoft.com/en-us/graph/api/message-send?view=graph-rest-1.0). You can also [move a message](https://docs.microsoft.com/en-us/graph/api/message-move?view=graph-rest-1.0) (from deleted items to inbox for example). But I don't believe you can just create a message in the inbox like that (without sending it). However the best place to get answers about API capabilities [is through the dedicated Microsoft Q&A forum](https://docs.microsoft.com/en-us/answers/topics/microsoft-graph-mail.html) which is monitored by the product team. (this repo issues/discussions is really about the SDK itself)

**Reply by dadams39** [reply]
SDK does claim that a message can be updated.
```
    requestBody := models.NewMessage()
    subject := "Cannot Do Worse"
    requestBody.SetSubject(&subject)
    resp := client.UsersById(user).MessagesById(id).Patch(requestBody)
    fmt.Printf("Response: %v\n", resp)
}
```
Update methods depend on Patch methods. From my short investigation, the response is nil AND the subject is not updated. Has the update function changed from what is located [here](https://docs.microsoft.com/en-us/graph/api/message-update?view=graph-rest-1.0&tabs=go)

**Reply by baywet** [reply]
no, the function is still [the same](https://github.com/microsoftgraph/msgraph-sdk-go/blob/80186093e26af8b4046b4eb89a342c4a8af0641c/users/item/messages/item/message_item_request_builder.go#L254)

**Comment by dadams39** [comment]
Asked in Issue #155 , but I will say that I have not found the ability to modify a message. An message can be moved to a different directory, but that appears to be it. I have also added the question to the Q&A forum, but I have not heard anything since.

**Reply by baywet** [reply]
I think you should be able to modify things like the categories or the flag information as well. But yes once the message has been sent/received, you can't change things like subject, body etc...

**Comment by dadams39** [comment]
If one cannot change update an old message via the graph API,  the early suggestion of creating a draft should work.
When I pull in an old message and try to setID to the new message Id and post it, I am receiving an error from the API.
Code here :

```go
parseNode, err := absser.DefaultParseNodeFactoryInstance.GetRootParseNode("application/json",bytes)
if err != nil{
        fmt.Errorf("Unable to create parseNode:%v\n", err)
    }
aMessage, err := parseNode.GetObjectValue(models.CreateMessageFromDiscriminatorValue)
if err != nil{
        fmt.Printf("Unable to create message: %v\n", err)
}
if aMessage == nil{
        os.Exit(1)
}
message := aMessage.(models.Messageable)
message.SetId( &id )
mailFolderId := "AQMkAGQ1NzViZTdhLTEwMTMtNGJjNi05YWI2LTg4ADVkZWQwNmNlMTgALgAAAw_9XBStqZdPuOVIalVTz7sBAHzBhzS2FNNNiXdGkRghSr4AAAIBDAAAAA=="
attrib, err := client.UsersById(user).MailFoldersById(mailFolderId).Messages().Post(message) // switch Me for userby id
```

**Reply by dadams39** [reply]
The error that I'm receiving is as follows:
`BadRequest Unable to read JSON request payload. Please ensure Content-Type header is set and payload is of valid JSON format. `
